### DISCUSSION TITLE
Initiative Assignment name is getting trimmed

### DISCUSSION THREAD

**Discussion by uaghani** [gettek/terraform-azurerm-policy-as-code#82] [Q&A]
I have an issue where the assignment display name is getting trimmed into only 24 characters which I believe is because of this 
![image](https://github.com/gettek/terraform-azurerm-policy-as-code/assets/55552196/19b38644-e396-4b25-8600-783f58b3d6e0)
I was able to resolve it by using a new input parameter for assignment_display_name and fixed in the module def_assignment.
I tried to resolve it in  set_assignment module for an initiative assignment but the output is same
Here is how my code looks like.

```
module "initiative_require_tags_r_rg" {
  source                  = "gettek/policy-as-code/azurerm//modules/initiative"
  initiative_name         = "Tag-Require Tags to be present-Initiative"
  initiative_display_name = "Require Tags Tags to be present-Initiative"
  initiative_description  = "Looks for tags Tags to be present on Resource and Resource Groups"
  initiative_category     = "Tagging"

  member_definitions = [
    module.policy-definitions["tag-tagname_existence_resource"].definition,
    module.policy-definitions["tag-tagname_existence_resourcegroup"].definition
  ]
  merge_parameters = true
}
```
```
module "assignment_require_tags_r_rg" {
  source           = "gettek/policy-as-code/azurerm//modules/set_assignment"
  initiative       = module.initiative_require_tags_r_rg
  assignment_scope = var.assignment_scope //Assignment Scope i.e. Resource Group ID or Management group ID

  for_each = toset([
    "Owner",
    "Environment"
  ])
  assignment_name         = "Tags-${each.key} must exist"
  assignment_display_name = "Tags-${each.key} must exist"
  assignment_description  = "Requires resources and resource groups ${each.key} tag value to be present"
  assignment_parameters = {
    tagName = each.key
  }
  skip_remediation = var.skip_remediation
}
```

**Comment by gettek** [comment]
The assignment modules adhere to [Azure limitations](https://github.com/gettek/terraform-azurerm-policy-as-code#limitations) but will need to check if and when these have changed. Going by the terraform docs;
- `azurerm_management_group_policy_assignment`: [`name` ](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/management_group_policy_assignment#name)_must be between 3 and 24 characters in length_
- All other assignments: [`name`](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/subscription_policy_assignment#name) _Cannot exceed 64 characters in length._

[`DisplayName`](https://github.com/gettek/terraform-azurerm-policy-as-code/blob/main/modules/definition/variables.tf#L24) on the hand has a maximum length of 128 characters and [`description`](https://github.com/gettek/terraform-azurerm-policy-as-code/blob/main/modules/definition/variables.tf#L35) a maximum length of 512 characters which are not trimmed but validated

**Reply by uaghani** [reply]
This is great! Simplified so much of my work with Az Policies. Thanks for your great work.
I was able to get it to work. I missed the part where the `initiative = name.initiative`  requires to be a reference to `.initiative` and not name.

**Reply by gettek** [reply]
FYI; assignment_name trims updated in [2.8.2](https://github.com/gettek/terraform-azurerm-policy-as-code/releases/tag/2.8.2)
